### PR TITLE
More vonage connect view to the channel type

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/channels/models.py [hunk 1]
```diff
@@ -284,6 +284,9 @@ class Channel(LegacyUUIDMixin, TembaModel, DependencyMixin):
     CONFIG_VONAGE_APP_ID = "nexmo_app_id"
     CONFIG_VONAGE_APP_PRIVATE_KEY = "nexmo_app_private_key"
 
+    SESSION_VONAGE_API_KEY = "VONAGE_API_KEY"
+    SESSION_VONAGE_API_SECRET = "VONAGE_API_SECRET"
```
Comment (rowanseymour) [inline temba/channels/models.py:288]: why do these need to be in `Channel` ?

File: temba/channels/types/vonage/tests.py [hunk 2]
```diff
@@ -156,15 +175,67 @@ def test_claim(self, mock_update_number, mock_buy_number, mock_get_numbers, mock
         self.assertContains(response, reverse("courier.nx", args=[channel.uuid, "status"]))
         self.assertContains(response, reverse("mailroom.ivr_handler", args=[channel.uuid, "incoming"]))
 
+    @patch("temba.channels.types.vonage.client.VonageClient.check_credentials")
+    def test_vonage_connect(self, mock_check_credentials):
+        self.login(self.admin)
+
+        connect_url = reverse("channels.types.vonage.connect")
+
+        self.assertFalse(Channel.SESSION_VONAGE_API_KEY in self.client.session)
```
Comment (rowanseymour) [inline temba/channels/types/vonage/tests.py:184]: `self.assertNotIn`

File: temba/channels/types/vonage/tests.py [hunk 3]
```diff
@@ -156,15 +175,67 @@ def test_claim(self, mock_update_number, mock_buy_number, mock_get_numbers, mock
...
+        # ok, now with a success
+        mock_check_credentials.return_value = True
+
+        response = self.client.post(connect_url, {"api_key": "key", "api_secret": "secret"})
+        self.assertEqual(response.status_code, 302)
+
+        response = self.client.post(connect_url, {"api_key": "key", "api_secret": "secret"}, follow=True)
+        self.assertEqual(response.request["PATH_INFO"], reverse("channels.types.vonage.claim"))
+
+        self.assertTrue(Channel.SESSION_VONAGE_API_KEY in self.client.session)
```
Comment (rowanseymour) [inline temba/channels/types/vonage/tests.py:214]: `self.assertIn`

File: templates/channels/types/vonage/claim.haml [hunk 4]
```diff
@@ -1,6 +1,15 @@
 -extends "channels/channel_claim_number.html"
 -load i18n
 
+- block account-trial-warning
```
Comment (rowanseymour) [inline templates/channels/types/vonage/claim.haml:4]: `-block`

feels a bit weird this is using `account-trial-warning`,  no?

**Review by rowanseymour — CHANGES_REQUESTED** [review]

File: templates/channels/types/vonage/claim.haml (hunk 4 above)
Comment (norkans7) [reply templates/channels/types/vonage/claim.haml:4]: I will rename that

File: temba/channels/types/vonage/tests.py (hunk 2 above)
Comment (norkans7) [reply temba/channels/types/vonage/tests.py]: updated

File: temba/channels/types/vonage/tests.py [hunk 5]
```diff
@@ -156,15 +176,67 @@ def test_claim(self, mock_update_number, mock_buy_number, mock_get_numbers, mock
...
         self.channel = Channel.create(
-            self.org, self.user, "RW", "NX", None, "+250788123123", uuid="00000000-0000-0000-0000-000000001234"
+            self.org,
+            self.user,
+            "RW",
+            "NX",
+            None,
+            "+250788123123",
+            uuid="00000000-0000-0000-0000-000000001234",
+            config={Channel.CONFIG_VONAGE_API_KEY: "1234", Channel.CONFIG_VONAGE_API_SECRET: "secret"},
```
Comment (rowanseymour) [inline temba/channels/types/vonage/tests.py:232]: can't these move into the channel package too?

Comment (norkans7) [reply temba/channels/types/vonage/tests.py]: Updated, some channels tests import those locally to avoid circular dependency issue

**Pull request merged**